py/kubeflow/testing/run_e2e_workflow.py

```diff
@@ -343,6 +350,86 @@ def run(args, file_handler): # pylint: disable=too-many-statements,too-many-bran
               "?tab=workflow".format(workflow_name))
       ui_urls[workflow_name] = ui_url
       logging.info("URL for workflow: %s", ui_url)
+    elif w.tekton_run:
+      tekton_run = None
+      with open(w.tekton_run) as config_f:
+        tekton_run = yaml.load(config_f)
+      if tekton_run.get("kind", "") != "PipelineRun":
+        logging.error("Invalid Tekton config kind: %s", tekton_run.get("kind", ""))
+        continue
+      test_target_name = tekton_run["metadata"].get("name", "")
+      if not test_target_name:
+        test_target_name = workflow_name
+      logging.info("Reading Tekton PipelineRun config: %s", test_target_name)
+      tekton_run["metadata"]["name"] = workflow_name
+
+      for t in tekton_run.get("spec", {}).get("pipelineSpec", {}).get("tasks", []):
+        if not "params" in t:
+          t["params"] = []
+
+        # Update Junit artifacts XML.
+        for i in range(len(t["params"])):
```

Comment:
Could you explain why run_e2e_workflow.py has to adjust the path of the junit files?